### PR TITLE
Replace data transfomer in some modules

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

File: starknet_py/contract.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def parsed_abi(self) -> Abi:
         return AbiParser(self.abi).parse()
 
     @staticmethod
-    def from_abi(address: int, abi: ABI) -> "ContractData":
+    def from_abi(address: int, abi: ABI) -> ContractData:
```
Comment (Solpatium) [inline starknet_py/contract.py:71]: We use __future__.annotations in this file, so we can use types normally.

File: starknet_py/net/models/abi/model.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
 from __future__ import annotations
 
-from collections import OrderedDict
 from dataclasses import dataclass
-from typing import Dict, Optional
+from typing import Dict, Optional, OrderedDict
```
Comment (Solpatium) [inline starknet_py/net/models/abi/model.py:4]: We need OrderedDict type here (not class) :|

**Review by drknzz — APPROVED** [review]
👍

File: starknet_py/utils/contructor_args_translator.py [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
 
 
 def translate_constructor_args(
     abi: List, constructor_args: Optional[Union[List, dict]]
 ) -> List[int]:
-    constructor_abi = next(
-        (member for member in abi if member["type"] == "constructor"),
-        None,
-    )
+    parsed = AbiParser(abi).parse()
```
Comment (war-in) [inline starknet_py/utils/contructor_args_translator.py:10]: ❤️

**Review by war-in — APPROVED** [review]

**Pull request merged**